storybook/src/components/Button/Button.test.stories.tsx

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import type { ButtonProps } from '@amsterdam/design-system-react'
+import type { Meta, StoryObj } from '@storybook/react-vite'
+
+import { CloseIcon } from '@amsterdam/design-system-react-icons'
+import { Button } from '@amsterdam/design-system-react/src'
+import { buttonVariants } from '@amsterdam/design-system-react/src/Button/Button'
+
+const meta = {
+  title: 'Components/Buttons/Button',
+  component: Button,
+  args: {
+    children: 'Versturen',
+    disabled: false,
+    icon: undefined,
+    iconBefore: false,
+    iconOnly: undefined,
+    variant: 'primary',
+  },
+} satisfies Meta<typeof Button>
+
+export default meta
+
+type Story = StoryObj<typeof meta>
+
+const states = ['default', 'disabled', 'hover', 'icon']
+
+export const Test: Story = {
+  parameters: {
+    pseudo: {
+      hover: ['.hover'],
+    },
+  },
+  render: () => (
+    <table>
```

Comment:
This is a bit of a nitpick, but: you're using a `table` here for layout purposes, which you generally shouldn't do. A `table` is for tabular data. 

It's ok here because we only use it for visual tests, but be aware that you should generally avoid this.